### PR TITLE
Initialize is  in separate function, fixed bug with gst_started_flag

### PR DESCRIPTION
**Pull request by PavelAtCarbyne**

### CODE REVIEW

File: plugins/janus_videoroom.c [hunk 1]
```diff
@@ -1786,6 +1786,12 @@ static void janus_videoroom_recorder_close(janus_videoroom_publisher *participan
  * @returns TRUE  in case of success, FALSE otherwise */
 gboolean wait_for_pipeline_close(forward_media_type media_type,
                                    janus_videoroom*  room );
+/*! \brief Method for test is pipeline processing (creation,running,...) already started , for prevent multiple start
+ * @param[in] media_type, enumerator of desired media type pipeline
+ * @param[in] room,  the videoroom plugin room object
+ * @returns TRUE  in case of pipeline started, FALSE otherwise */
+gboolean is_pipeline_processing_started(forward_media_type media_type,
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: add static

File: plugins/janus_videoroom.c [hunk 2]
```diff
@@ -1786,6 +1786,12 @@ static void janus_videoroom_recorder_close(janus_videoroom_publisher *participan
  * @returns TRUE  in case of success, FALSE otherwise */
 gboolean wait_for_pipeline_close(forward_media_type media_type,
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: add static

File: plugins/janus_videoroom.c [hunk 3]
```diff
@@ -2198,6 +2204,22 @@ static void janus_videoroom_rtp_forwarder_free(const janus_refcount *f_ref) {
...
+	if(NULL == videoroom) {
+		return;
+	}
+	for (int media_type_counter=0; media_type_counter < FORWARD_MEDIA_TYPE_SIZE; media_type_counter ++ ) {
+		videoroom->gst_thread_parameters[media_type_counter].gstr.pipeline = NULL;
+		videoroom->gst_thread_parameters[media_type_counter].gstr.m_mainLoop = NULL;
+		videoroom->gst_thread_parameters[media_type_counter].gstr.m_watchID = 0;
+		g_atomic_int_set(&videoroom->gst_thread_parameters[media_type_counter].gstr.gst_started_flag, 0);
+		g_atomic_int_set(&videoroom->gst_thread_parameters[media_type_counter].gstr.gst_defined_flag, 0);
+		g_atomic_int_set(&videoroom->gst_thread_parameters[media_type_counter].gst_run_flag,0);
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: logstr init => memset (0, MAX_STRING_LEN)

Comment (alex-dizengof) [reply plugins/janus_videoroom.c]: media_type = 0

File: plugins/janus_videoroom.c [hunk 4]
```diff
@@ -5067,6 +5073,23 @@ gboolean wait_for_pipeline_close(forward_media_type media_type,
 	return TRUE;
 }
 
+gboolean is_pipeline_processing_started(forward_media_type media_type,
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: start_pipline_processing_flag
TRUE -> success and start pipeline
FALSE -> fail. already running. ignore

File: plugins/janus_videoroom.c [hunk 5]
```diff
@@ -5084,11 +5107,7 @@ gboolean forward_media(janus_videoroom_session *session, publisher_media_type  m
         if (participant->room) {
                 janus_mutex_lock(&participant->room->mutex);
                 if(PUBLISHER_MEDIA_VIDEO == media_type ) {
-
-                	if(g_atomic_int_get(&participant->room->gst_thread_parameters[MEDIA_VIDEO].gstr.gst_started_flag)) {
-                        	JANUS_LOG(LOG_ERR, "CARBYNE:::: VIDEO pipeline already running  room:%s port:%d\n",
-                                	                participant->room_id_str,
-                                        	        participant->room->gst_thread_parameters[MEDIA_VIDEO].forward_port_1);
+                	if(is_pipeline_processing_started(MEDIA_VIDEO, participant->room)) {
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: flip if case

File: plugins/janus_videoroom.c [hunk 6]
```diff
@@ -5135,14 +5154,11 @@ gboolean forward_media(janus_videoroom_session *session, publisher_media_type  m
...
                 if(PUBLISHER_MEDIA_AUDIO == media_type) {
-                	if(g_atomic_int_get(
-				&participant->room->gst_thread_parameters[AUDIO_FORWARD_MEDIA_TYPE_FROM_BOOL(participant->is_ingress)].gstr.gst_started_flag)) {
-                        	JANUS_LOG(LOG_ERR, "CARBYNE:::: AUDIO pipeline already running %s port:%d\n",
-                                                   participant->room_id_str,
-                                                   participant->room->gst_thread_parameters[AUDIO_FORWARD_MEDIA_TYPE_FROM_BOOL(participant->is_ingress)].forward_port_1);
-                        	janus_mutex_unlock(&participant->room->mutex);
-                        	return FALSE;
-                	}
+			if(is_pipeline_processing_started(AUDIO_FORWARD_MEDIA_TYPE_FROM_BOOL(participant->is_ingress), participant->room)) {
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c]: flip if case

File: plugins/janus_videoroom.c [hunk 7]
```diff
@@ -5171,6 +5187,9 @@ gboolean forward_media(janus_videoroom_session *session, publisher_media_type  m
                                         participant->room_id_str,
                                         participant->room->gst_thread_parameters[MEDIA_AUDIO_MIXER].forward_port_1,
                                         participant->room->gst_thread_parameters[MEDIA_AUDIO_MIXER].forward_port_2);
+
+				g_atomic_int_set(&participant->room->gst_thread_parameters[MEDIA_AUDIO_MIXER].gstr.gst_started_flag, 1);
```
Comment (alex-dizengof) [inline plugins/janus_videoroom.c:5194]: add comment why we do it here

**Review by alex-dizengof — CHANGES_REQUESTED** [review]

File: plugins/janus_videoroom.c (hunk 7 above)
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c:5194]: done

**Review by alex-dizengof — APPROVED** [review]

**Pull request merged**